### PR TITLE
HW-262: Styling mailing schedule page

### PR DESCRIPTION
**Pull request by meladawy**
1- Changes in styling to match the target design for "New Mailing > Options" page
2- Install Gulp and Sass for the project
3- Change Wizard class naming and define it as a new CRM Component following CSS BEM naming standard

Before:
screen shot 2017-03-29 at 3 45 55 pm

After:
screen shot 2017-03-29 at 3 46 58 pm

### CODE REVIEW

File: sass/mosaico/_form.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.crm-mosaico-page {
+  #inputSubject {
+    width: 80%;
+  }
+
+  input[name=campaign] {
+    width: 98%;
+  }
+
+  select {
```
Comment (boriszaydlin) [inline sass/mosaico/_form.scss]: please include name values into quotes `&[name="optout_id"]`
same below and above

File: css/legacy.css [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#bootstrap-theme .crm-mosaico-page .helpicon{background-color:#42afcb}
```
Comment (boriszaydlin) [inline css/legacy.css]: This is not same as here https://github.com/compucorp/uk.co.vedaconsulting.mosaico/pull/4/files#diff-1e6bdecc7a4b9016c394ff3e70c9209dL2
You added space between this two selectors `#bootstrap-theme .crm-mosaico-page`, check if we really need this

File: sass/civicrm-override/4.6.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// This file should be called only if the current version is 4.6
```
Comment (boriszaydlin) [inline sass/civicrm-override/4.6.scss]: It is import file now, don't we need to add `_` for filename: `_4.6.scss` ?

File: sass/mosaico/_form.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.crm-mosaico-page {
+  #inputSubject {
+    width: 80%;
+  }
+
```
Comment (boriszaydlin) [inline sass/mosaico/_form.scss:5]: Please use quotes `[name="campaign"]`, same below

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      exclude: [/^html/, /^body/]
+    })]))
+    .pipe(cssnano())
+    .pipe(sourcemaps.write('./'))
+  .pipe(gulp.dest('./css/'))
+});
+
+gulp.task('sass', ['sass:main','sass:legacy']);
+
+gulp.task('watch', function () {
```
Comment (boriszaydlin) [inline gulpfile.js]: use two spaces tab indent

File: gulpfile.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      browsers: ['last 2 versions'],
+      cascade: false
+    }))
+    .pipe(postcss([postcssPrefix({
+      prefix: bootstrapNamespace + ' ',
+      exclude: [/^html/, /^body/]
+    })]))
+    .pipe(cssnano())
+    .pipe(sourcemaps.write('./'))
+  .pipe(gulp.dest('./css/'))
```
Comment (AkA84) [inline gulpfile.js]: the two tasks "bodies" are basically the same, only the `src()` argument is different. So you should unify them in, say, the same function

File: sass/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+$crm-input-border:                        #c2cfd8;
+
+// Mosaico Related Variables
+$crm-mosaico-template-item:               #f1f5f6;
+$crm-mosaico-template-border:             #e8eef0;
+$crm-mosaico-template-item-wrapper:       #333;
+$crm-mosaico-template-item-shadow:        inset 0 1px 10px 0 rgba(0, 0, 0, .25);
+
+// Identity Colors
+$brand-primary:                           #42AFCB;
```
Comment (AkA84) [inline sass/_variables.scss]: this is a bootstrap variable, why is it needed here?

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: sass/mosaico/_form.scss (hunk 4 above)
Comment (meladawy) [reply sass/mosaico/_form.scss:5]: Fixed in a new commit 
https://github.com/compucorp/uk.co.vedaconsulting.mosaico/pull/4/commits/5635e5d993ba9e502acf3d9a9c0d0c75d3b32dca

File: sass/_variables.scss (hunk 7 above)
Comment (meladawy) [reply sass/_variables.scss]: Its required by this file 
https://github.com/compucorp/uk.co.vedaconsulting.mosaico/pull/4/files#diff-a9e67b892cd069c8d017d3e8687d28f4

I get "Variable not defined" error if i removed it from _variables.scss

**Comment by meladawy** [comment]
@AkA84 I installed CivicrmSassRoot and replaced local variables with shoreditsh variables 

https://github.com/compucorp/uk.co.vedaconsulting.mosaico/pull/4/commits/6076ccb8359fb1f60abe278326b6190d5e790316

**Review by AkA84 — APPROVED** [review]

**Review by boriszaydlin — APPROVED** [review]

**Pull request merged**